corehq/apps/users/user_data.py

```diff
@@ -60,12 +67,32 @@ def save(self):
 
     @property
     def _provided_by_system(self):
-        return {
+        provided_data = {
             **(self.profile.fields if self.profile else {}),
             PROFILE_SLUG: self.profile_id or '',
             COMMCARE_PROJECT: self.domain,
         }
 
+        def _add_location_data():
+            if self._couch_user.get_location_id(self.domain):
+                provided_data[COMMCARE_LOCATION_ID] = self._couch_user.get_location_id(self.domain)
+                provided_data[COMMCARE_PRIMARY_CASE_SHARING_ID] = self._couch_user.get_location_id(self.domain)
+
+            if self._couch_user.get_location_ids(self.domain):
+                provided_data[COMMCARE_LOCATION_IDS] = user_location_data(
+                    self._couch_user.get_location_ids(self.domain))
+
+        # Some test don't have an actual user existed
+        # Web User don't store location fields in user data
+        if self._couch_user and self._couch_user.is_commcare_user() or not settings.UNIT_TESTING:
+            _add_location_data()
```

Comment:
I don't think this conditional is quite right, if the user is a web user, this conditional will be: `if True and False or True`, which evaluates to `True`, meaning we'll attempt to add location data.
```suggestion
        # Some test don't have an actual user existed
        # Web User don't store location fields in user data
        if (self._couch_user or not settings.UNIT_TESTING) and self._couch_user.is_commcare_user():
            _add_location_data()
```